src/client/src/App.jsx

```diff
@@ -0,0 +1,79 @@
+// #region imports
+import { useEffect, useMemo, useState } from 'react';
+import './App.css'
+import { MessageForm } from './MessageForm.jsx';
+import { MessageList } from './MessageList.jsx';
+import { apiUtils } from './API/apiUtils.js';
+import { Room } from './Room.jsx';
+// #endregion
+
+const DataLoader = ({ onData, room, userName }) => {
+  const WS_URL = 'ws://localhost:3005';
+  useEffect(() => {
+    const socket = new WebSocket(WS_URL);
+
+    socket.addEventListener('open', () => {
+      socket.send(JSON.stringify(
+        { text: `${userName} join the room`, userName, room, type: 'join' }
+      ));
+    });
+
+    socket.addEventListener('message', (event) => {
+      const text = JSON.parse(event.data)
+
+      onData(text);
+    });
+
+    return () => {
+      socket.close();
+    };
+  }, [room])
+
+  return (
+    <h1 className="title">Chat application</h1>
+  );
+};
+
+export function App() {
+  const [messages, setMessages] = useState([]);
+  const [userName, setUserName] = useState(null);
+  const [room, setRoom] = useState('room1');
+
+  const sortedMessages = useMemo(() => {
+    return messages.filter(message => message.room === room);
+  },[messages, room]);
+
+  useEffect(() => {
+    if (!localStorage.getItem('name')) {
+      let name = '';
+
+      while (!name) {
+        name = prompt('Please enter your name');
+      }
+
+      apiUtils.post({
+        userName: name, room, type: 'join'
+      })
+
+      setUserName(name);
+      localStorage.setItem('name', name);
+    } else {
+      const name = localStorage.getItem('name');
+      setUserName(name)
+    }
+  }, [room]);
+
+  function saveData(message) {
+    setMessages((messages) => { return [...messages, message] })
```

Comment:
The 'saveData' function can be simplified to use an implicit return, like so: setMessages(messages => [...messages, message])